restli-tools/src/main/resources/apiVmTemplates/action.vm

```diff
@@ -0,0 +1,114 @@
+#*
+Copyright (c) 2021 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*#
+
+  #foreach($method in $spec.actions)
+    #set($actionParamClassName = "${util.nameCapsCase($method.name)}ActionParameters")
+    #doc($method.schema.doc)
+
+    ## methods are named as "action<ActionName>" so it is consistent with current RequestBuilder naming
+    #if(${method.hasActionParams()})
+    @SuppressWarnings("unchecked")
+    public CompletionStage<${method.valueClassDisplayName}> action${util.nameCapsCase(${method.name})}(
+      #if(${method.isEntityAction()})
+      $spec.keyClassName $spec.idName,
+      #end
+        #set($paramLen = ${method.getParameters().size()})
+        #set($lastParam = ${method.getParameters().get($paramLen - 1)})
+        #if($paramLen > 1)
+          #foreach($idx in [2..$paramLen])
+            #set($param = ${method.getParameters().get($idx - 2)})
```

Comment:
To be consistent with Rest method behavior, we would have to add only the required parameters here.

Reply:
Sorry I don't understand this comment. I am only adding the "actionParam" here. What do you mean by 
"add only the required parameters here"

Reply:
ActionParams can be optional or required. In the fluent api, only the required parameters are directly specified. All optional parameters go into a xxxOptionalParams class. This is so we can add new optional parameters to action methods without breaking fluent APIs. 

This is already done for REST methods.